registrations/views.py

```diff
@@ -363,6 +363,64 @@ def get(self, request, *args, **kwargs):
         return Response(resp, status=status)
 
 
+class JembiFacilityCheckHealthcheckView(APIView):
+
+    """ Jembi Facility Check Healthcheck Interaction
+        GET - returns service up - getting auth'd requires DB
+    """
+
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, *args, **kwargs):
+        if not (
+            settings.JEMBI_BASE_URL
+            and settings.JEMBI_USERNAME
+            and settings.JEMBI_PASSWORD
+        ):
+
+            return Response(
+                "Jembi integration is not configured properly.",
+                status.HTTP_503_SERVICE_UNAVAILABLE,
+            )
+
+        try:
+            clinic_code = request.query_params.get("clinic_code")
+            result = requests.get(
+                urljoin(settings.JEMBI_BASE_URL, "NCfacilityCheck"),
+                headers={"Content-Type": "application/json"},
+                auth=(settings.JEMBI_USERNAME, settings.JEMBI_PASSWORD),
+                params={"criteria": "value:{}".format(clinic_code)},
+                verify=False,
+            )
+            result.raise_for_status()
+
+            jembi_result = result.json()
+
+            if len(jembi_result.get("rows")) > 0:
+                resp = {
+                    "up": True,
+                    "result": {"Facility": jembi_result.get("rows")[0][2]},
+                }
+            else:
+                raise Http404()
```

Comment:
@rudigiesler just a check on this - should we return 404 for when the service is down? or is there a better error code?

Reply:
@miltontony I think there are 2 codes here depending on what the failure is.
502 - we received a bad response from upstream
504 - we timed out requesting from upstream

Reply:
Looking at this again, I think a 502 error is more suited here